zerver/tests/test_user_topics.py

```diff
@@ -114,11 +114,11 @@ def test_add_muted_topic(self) -> None:
         assert stream.recipient is not None
         result = self.api_patch(user, url, data)
 
-        # Now check that error is raised when attempted to mute an already
-        # muted topic. This should be case-insensitive.
+        # Now check that no error is raised when attempted to mute
+        # an already muted topic. This should be case-insensitive.
         data["topic"] = "VERONA3"
         result = self.api_patch(user, url, data)
-        self.assert_json_error(result, "Topic already muted")
+        self.assert_json_success(result)
```

Comment:
Given the comment about case-sensitivity, we should really extend this test to verify that we didn't end up with duplicate UserTopic rows with the two different cases.

Reply:
[Extended](https://github.com/zulip/zulip/pull/24690/commits/d5d5cbe378abe3fd3f1c14c366dd60a7ce6594c0#diff-dca29140ac4182499e88c086202b897cca41c670b6163288b76f582569528402R136) the test to verify.